bitnami/spark/templates/_helpers.tpl

```diff
@@ -31,6 +31,27 @@ the service name.
 {{- end }}
 {{- end -}}
 
+{{/*
+Return the secret containing Spark TLS certificates
+*/}}
+{{- define "spark.tlsSecretName" -}}
+{{- $secretName := .Values.security.certificatesSecretName -}}
+{{- if $secretName -}}
+    {{- printf "%s" (tpl $secretName $) -}}
+{{- else -}}
+    {{- printf "%s-crt" (include "common.names.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a TLS secret object should be created
+*/}}
+{{- define "spark.createTlsSecret" -}}
+{{- if and .Values.security.autoGeneratedCertificates (not .Values.security.certificatesSecretName) }}
```

Comment:
I think we should only check the 1st condition:
```suggestion
{{- if .Values.security.autoGeneratedCertificates }}
```
Then, you can add a validation to thrown an error when both `.Values.security.autoGeneratedCertificates` and `.Values.security.certificatesSecretName` are set